tutorials/llm/nemo2-peft.ipynb

```diff
@@ -0,0 +1,2052 @@
+    "\n",
+    "- Modular Abstractions - By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 simplifies adaptation and experimentation. This modular approach allows developers to more easily modify and experiment with different components of their models.\n",
+    "\n",
+    "- Scalability - NeMo 2.0 seamlessly scales large-scale experiments across thousands of GPUs using NeMo-Run, a powerful tool designed to streamline the configuration, execution, and management of machine learning experiments across computing environments.\n",
+    "\n",
+    "By adopting PyTorch Lightning’s modular abstractions, NeMo 2.0 makes it easy for users to adapt the framework to their specific use cases and experiment with various configurations. This section offers an overview of the new features in NeMo 2.0 and includes a migration guide with step-by-step instructions for transitioning your models from NeMo 1.0 to NeMo 2.0.\n",
+    "\n",
+    "## NeMo-Run\n",
+    "\n",
+    "NeMo-Run is a powerful tool designed to streamline the configuration, execution, and management of machine learning experiments across various computing environments. To run your experiments with Nemo-Run, you need to follow the three steps:\n",
+    "1. Configure your function\n",
+    "\n",
+    "2. Define your Executor\n",
+    "\n",
+    "3. Run your experiment\n",
+    "\n",
+    "We will be using NeMo-Run to run our experiments in this notebook.\n",
+    "\n",
+    "\n",
+    "# NeMo Tools and Resources\n",
+    "1. [NeMo Github repo](https://github.com/NVIDIA/NeMo)\n",
+    "\n",
+    "2. [NeMo-Run Github repo](https://github.com/NVIDIA/NeMo-Run/)\n",
+    "\n",
+    "3. NeMo Framework Training container: `nvcr.io/nvidia/nemo:dev`\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Educational Resources\n",
+    "1. Blog: [Mastering LLM Techniques: Customization](https://developer.nvidia.com/blog/selecting-large-language-model-customization-techniques/)\n",
+    "\n",
+    "2. Whitepaper: [LoRA: Low-Rank Adaptation of Large Language Models](https://arxiv.org/abs/2106.09685)\n",
+    "\n",
+    "3. [NeMo 2.0 Overview](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/index.html)\n",
+    "\n",
+    "4. Blog: [Tune and Deploy LoRA LLMs with NVIDIA TensorRT-LLM](https://developer.nvidia.com/blog/tune-and-deploy-lora-llms-with-nvidia-tensorrt-llm/)\n",
+    "\n",
+    "\n",
+    "## Software Requirements\n",
+    "\n",
+    "1. Use the latest [NeMo Framework Training container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo/tags) . Note that you must be logged in to the container registry to view this page.\n",
+    "\n",
+    "2. This notebook uses the container: `nvcr.io/nvidia/nemo:dev`.\n",
+    "\n",
+    "\n",
+    "## Hardware Requirements\n",
```

Comment:
Can we please move any requirements outside the notebooks in a separate README. The README should also contain the instructions to launch the notebook.

Example: https://github.com/NVIDIA/NeMo/tree/main/tutorials/llm/llama-3/biomedical-qa

It needn't be very comprehensive, just what the user needs to know indicated clearly.

For any information (like features of NeMo 2.0 and supported models), always point to the documentation.

Reply:
Do I create a separate folder for these notebooks so that I can put a README in that folder?

Reply:
Moved to README. Thanks!